src/libraries/System.Net.Security/src/System/Net/Security/Pal.OSX/SafeDeleteSslContext.cs

```diff
@@ -158,13 +169,24 @@ private unsafe int WriteToConnection(void* connection, byte* data, void** dataLe
             ulong length = (ulong)*dataLength;
             Debug.Assert(length <= int.MaxValue);
 
+            if (Interlocked.Exchange(ref _pendingOutput, 1) == 2)
```

Comment:
Seems like these should use CompareExchange rather than Exchange.  Otherwise, while this first call will be able to use the returned previous value to see whether it was for disposal, a subsequent caller will see the value as 1 instead of 2.

Reply:
yes. should be fixed now. I realized that when working on the name change. In practice the managed and unmanaged pairs should not be called at the same time but I don't want this logic to depend on it.